client/src/extension.ts

```diff
@@ -129,11 +128,20 @@ export function activate(context: ExtensionContext) {
             searchProvider.renderSearchResult(searchResult);
         });
 
+        client.onNotification("vscoq/proofView", (proofView: ProofViewNotification) => {
+            const editor = window.activeTextEditor ? window.activeTextEditor : window.visibleTextEditors[0];
+            GoalPanel.proofViewNotification(context.extensionUri, editor, client, proofView);
+        });
+
         let goalsHook = window.onDidChangeTextEditorSelection(
             (evt: TextEditorSelectionChangeEvent) => {                    
-                const reqParams = makeCursorPositionUpdateProofViewRequestParams(evt);
-                if(reqParams !== null) {
-                    GoalPanel.refreshGoalPanel(context.extensionUri, evt.textEditor, client, reqParams);
+                if (evt.textEditor.document.languageId === "coq" 
+                    && workspace.getConfiguration('vscoq.proof').mode === 1
+                    && isMouseOrKeyboardEvent(evt)) 
+                {
+                    const textDocument = makeVersionedDocumentId(evt.textEditor); 
```

Comment:
Isn't this code duplicated from `sendInterpretToPoint` in `manualChecking`? Can't it be factorized?

Reply:
You are right. Just amended the code.